doc/app-METHOD.md

```diff
@@ -0,0 +1,80 @@
+<!---
+ Copyright (c) 2016 StrongLoop, IBM, and Express Contributors
+ License: MIT
+-->
+
+<h3 id='app.METHOD'>app.METHOD(path, callback [, callback ...])</h3>
+
+Routes an HTTP request, where METHOD is the HTTP method of the request, such as GET,
+PUT, POST, and so on, in lowercase. Thus, the actual methods are `app.get()`,
+`app.post()`, `app.put()`, and so on.  See below for the complete list.
+
+For more information, see the [routing guide](/guide/routing.html).
```

Comment:
>  I'm just doing what I was asked to do, and no one articulated the goal in general (to me anyway).

Right. One could say that this pull request was not even defined enough to begin working on, I understand the confusion. AFAIK doing this was only mentioned in passing, but no thought was really put into what that would mean, what it would look like, etc.

> I've wondered that myself, since I became involved in the project. IMHO that would be the best solution, but no one has actually proposed it until now. Having the API doc generated from comments directly in the code is ideal, because it helps to ensure that whenever someone updates the code, the doc is right there in their face.... 

Right. This has been proposed quite a few times, mainly before you came on board. Since there has never been any kind of official on-boarding, I wouldn't expect you to even know the history of the project, past proposal, etc. :)

> But there are other "pros and cons" to that approach: for example it's not clear we could make the API docs look as nice as they do now (at least without a lot of work). 

That is correct, all of which was taking into consideration a long time ago. That is how we got to where we are today: a separate website repository with nice human-friendly documentation. The latest major version of Jekyll can now directly render GH-style Markdwon, but the old one could not, making a website-specific repository almost the only viable solution.